app/controllers/api/v1/users_controller.rb

```diff
@@ -3,21 +3,81 @@ module V1
     class UsersController  < ApplicationController
       skip_before_action :verify_authenticity_token, if: :json_request?
       skip_before_filter  :verify_authenticity_token
-    def create
-      user = User.new(user_params)
-      if user.save
-        render json: { message: 'Thanks for signing up!' }, status: :created
-      else
-        render json: { error: user.errors }, status: :not_acceptable
+
+      def create
+        user = User.new(user_params)
+        if user.save
+          render json: { message: 'Thanks for signing up!' }, status: :created
+        else
+          render json: { error: user.errors }, status: :not_acceptable
+        end
       end
-    end
 
-    def user_params
-      params.require(:user).permit(:first_name, :last_name, :email, :password, :password_confirmation, :avatar)
-    end
+      def update
+        user = User.find(params[:id])
+        if user.update_attributes(user_params)
+          render json: { message: 'Your profile has been updated' }, status: :ok
+        else
+          render json: { error: user.errors }, status: :not_acceptable
+        end
+      end
 
-    end
+      def show
+        user = User.find(params[:id])
+        if user
+          data = {
+            'first_name' => user.first_name,
+            'last_name' => user.last_name,
+            'email' => user.email,
+            'avatar' => user.avatar
+          }
+          render json: { data: data }, status: :ok
+        else
+          render json: { message: user.errors }, status: :not_found
+        end
+      end
 
-  end
+      def destroy
+        if User.find(params[:id]).destroy
+          render json: { message: 'Your account is deleted with all your datas' }, status: :ok
+        else
+          render json: { message: user.errors }, status: :not_found
+        end
+      end
 
+      def forgot_password
+        user = User.find_by_email!(params[:user][:email])
+        user.send_password_reset if user
+        render json: { message: 'Password recovery link has been sent to your email' }, status: :ok
+      end
+
+      def change_password
```

Comment:
This method is very crowded, you shouldn't make controllers heavy, even I can't figure out what you are trying to accomplish here, please rewrite this whole method.